tests/integration/test_federation.py

```diff
@@ -0,0 +1,215 @@
+from copy import deepcopy
+import pytest
+import random
+
+import bigchaindb
+from bigchaindb.core import Bigchain
+from contextlib import contextmanager
+from bigchaindb.common.crypto import generate_key_pair
+from tests.pipelines.stepping import create_stepper
+
+
+################################################################################
+# Test setup code
+
+
+@contextmanager
+def federation(n):
+    """
+    Return a list of Bigchain objects and pipeline steppers to represent
+    a BigchainDB federation
+    """
+    keys = [generate_key_pair() for _ in range(n)]
+    config_orig = bigchaindb.config
+
+    @contextmanager
+    def make_nodes(i):
+        nonlocal keys
+        if i == 0:
+            yield []
+        else:
+            config = deepcopy(config_orig)
+            keys = [keys[-1]] + keys[:-1]
+            config['keypair']['private'] = keys[0][0]
+            config['keypair']['public'] = keys[0][1]
+            config['keyring'] = list(list(zip(*keys[1:]))[1])
```

Comment:
this would be more readable with a single list comprehension `[k.public_key for k in keys[1:]]`